explore/src/main/scala/explore/targeteditor/AladinCell.scala

```diff
@@ -92,41 +98,43 @@ object AladinCell extends ModelOptics {
       .useStateView(false)
       // to get faster reusability use a serial state, rather than check every candidate
       .useSerialState(List.empty[GuideStarCandidate])
+      // Analysis results
+      .useSerialState(List.empty[AgsAnalysis])
+      // Ags state
+      .useState[AgsState](AgsState.Idle)
       // Listen on web worker for messages with catalog candidates
-      .useStreamResourceBy((props, _, _, _, _) => props.tid)((props, _, _, _, gs) =>
-        _ =>
-          props.ctx.worker.streamResource.map(
-            _.map(decodeFromTransferable[WorkerMessage])
-              .filter {
-                case Some(CatalogResults(_) | CatalogQueryError(_)) => true
-                case _                                              => false
-              }
-              .collect {
-                case Some(r @ CatalogResults(_)) => fs2.Stream.emit[IO, CatalogResults](r)
-                case Some(CatalogQueryError(m))  =>
-                  fs2.Stream.raiseError[IO](new RuntimeException(m))
-              }
-              .flatten
-              .map(_.candidates.map { gsc =>
-                // We keep locally the data already pm corrected for the viz time
-                // If it changes over a month we'll request the data again and recalculate
-                // This way we avoid recalculatinng pm for example if only pos angle or
-                // conditions change
-                gsc.at(props.obsConf.vizTime)
-              })
-              .evalMap(r => gs.setStateAsync(r))
-          )
+      .useStreamResourceBy((props, _, _, _, _, _, _) => props.tid)(
+        (props, _, _, _, gs, ags, agsState) =>
+          _ =>
+            props.ctx.worker.streamResource.map(
+              _.map(decodeFromTransferable[WorkerMessage])
+                .filter {
+                  case Some(CatalogResults(_) | CatalogQueryError(_) | AgsResult(_)) => true
+                  case _                                                             => false
+                }
+                .evalMap {
+                  case Some(CatalogResults(candidates)) =>
```

Comment:
The `filter` didn't seem necessary with the `collect`, but there doesn't seem to be an `evalCollect`.

Maybe `.collect { ... }.evalMap(_)` avoids the filter?